### ISSUE TITLE
How to configure installment payments?

### ISSUE THREAD

**Issue by lynna-spidersilk** [compucorp/uk.co.compucorp.membershipextras#234]
More importantly than the other issue I posted: the main reason I installed this extension was to have monthly instalment payments for annual memberships. But having installed it now, I can't see any way of configuring that. 

The Edit Membership Type screen seems unchanged — there are no new options there. And there do not appear to be any new options under the Memberships or Administer > CiviMember menus either.

There is one new item under the main Administer menu called Payment Plan Options, but that does not contain any settings for instalment payments. The only options there are:

- Use latest price when auto renew membership?
- Offline payment processor for back office
- Days to renew in advance
- Custom groups to be excluded when auto-renew

No reference to instalment payments at all. So how does one configure that? Or is that feature not actually available yet?

**Comment by jamienovick** [comment]
@lynna-spidersilk - thanks for trying out membership extras!

Ok, point 1: we're just in the process of working out a number of bugs and a new v2.0 is imminent (next 2 weeks or so) so I would recommend either staying up to date with the master branch or waiting for that.

Point 2: The instalments are created when you add a new memberships on a per membership basis. It's not specific to a membership type so the same type can be used and paid with or without instalments. Go to add a new membership and you will get a new tab called "payment plan". Here you can setup a membership with instalments. See our draft user guide below (needs a bit of an update and we will post in CiviCRM gitbook eventually):

https://docs.google.com/document/d/1pabX0lyp8U90QsYVIaT2M0yTxztT9e_PnTeqebLdf_k/edit?usp=sharing

Also - are you using Drupal or Wordpress or Joomla? With Drupal there is also a suite of add on modules to support setting up payment plans with instalments from online forms

**Comment by lynna-spidersilk** [comment]
Thanks for the reply! I hadn't though of looking on the individual New Membership page — I had expected it to be configurable on the Membership Type page. 

I'm using Drupal, and I was hoping to use a Webform for the public-facing membership signup form (because CiviCRM's default forms for things tend to be overly complex and not very user-friendly, and ease of use is a big priority on this site), so yay for there being a module for that! 

But one question — the organization I'm working on this site for would like their memberships to be paid in monthly installments by default, rather than just as an option. Will it be possible to set up a webform to do that? 

I know, if they want people paying monthly it would be easier to just make the membership term 1 month instead of 1 year, but unfortunately setting policy is not up to me — I just have to find ways of implementing what they want. And they have reasons for wanting people to sign up for 1-year or 2-year terms, and offer them different bonuses for doing so...

**Comment by omarabuhussein** [comment]
I am late to the party, but if any one is looking on this issue, the answer for 

> But one question — the organization I'm working on this site for would like their memberships to be paid in monthly installments by default, rather than just as an option. Will it be possible to set up a webform to do that?
> 
> I know, if they want people paying monthly it would be easier to just make the membership term 1 month instead of 1 year, but unfortunately setting policy is not up to me — I just have to find ways of implementing what they want. And they have reasons for wanting people to sign up for 1-year or 2-year terms, and offer them different bonuses for doing so...

is Yes, you can configure a webform to create a one year membership by default to be paid in 12 installments (1 installment each month). but this module should also need to be installed https://github.com/compucorp/webform_civicrm_membership_extras for it to work properly .

**Comment by shaneonabike** [comment]
This documentation is amazing thank you so much! So to be clear there is no way in the actual _membership contribution page_ to add this instalment feature like you can for donations? This is really what I was looking for to see if our members needed that they could sign up for it.

**Comment by netzih** [comment]
> This documentation is amazing thank you so much! So to be clear there is no way in the actual _membership contribution page_ to add this instalment feature like you can for donations? This is really what I was looking for to see if our members needed that they could sign up for it.

Yes,  I am in the same situation, Especially since I use WordPress I would like those signing up online to have an option to pay in installments.
Is that something that is possible?

Thanks

**Comment by shaneonabike** [comment]
Depending on what payment processor you are using we had to patch our payment processor(s) in several situations to make this work properly. In one situation, iATS needed to be modified in order to properly handle instalments.

Are you able to do that on your own? If not, you could contact us to get some help.

cc @netzih

**Comment by omarabuhussein** [comment]
> This documentation is amazing thank you so much! So to be clear there is no way in the actual membership contribution page to add this instalment feature like you can for donations? This is really what I was looking for to see if our members needed that they could sign up for it.

Yeah unfortunately the work in this extension support contribution pages, For Drupal sites you can use webforms instead, but I am afraid that there is no alternative at the moment for Wordpress sites.
